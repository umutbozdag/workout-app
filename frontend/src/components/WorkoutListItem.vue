<template>
  <div class="workout-list-item" @click="handleOnItemClick(workoutId)">
    <div class="workout-list-item__info">
      <h3 class="workout-list-item__title">{{ title }}</h3>
      <p class="workout-list-item__desc">{{ description }}</p>
      <p v-if="completed && exerciseCount" class="workout-list-item__completed">
        Workout is completed
      </p>

      <div class="workout-list-item__exercise-count">
        <template v-if="exerciseCount === 0"> No exercises found. </template>
        <template v-if="exerciseCount > 1">
          {{ exerciseCount }} Exercises
        </template>
        <template v-if="exerciseCount === 1">
          {{ exerciseCount }} Exercise
        </template>
      </div>
    </div>

    <i class="el-icon-right workout-list-item__icon"></i>
  </div>
</template>

<script lang="ts">
import Vue from "vue";
export default Vue.extend({
  name: "WorkoutListItem",
  props: {
    workoutId: {
      type: String,
      default: "",
    },
    title: {
      type: String,
      default: "",
    },
    description: {
      type: String,
      default: "",
    },
    exerciseCount: {
      type: Number,
      default: 0,
    },
    completed: {
      type: Boolean,
      default: false,
    },
  },
  methods: {
    handleOnItemClick(workoutId: string) {
      this.$router.push({
        name: "WorkoutDetail",
        params: {
          workoutId,
        },
      });
    },
  },
});
</script>

<style lang="scss" scoped>
.workout-list-item {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 12px;
  border-radius: 6px;
  box-shadow: 2px 3px $primary;
  width: 620px;
  transition: 0.3s;
  cursor: pointer;
  border: 2px solid $primary;

  &:hover {
    box-shadow: 4px 6px $primary;
  }

  &__title {
    color: $primary-text;
    margin-top: 6px;
  }

  &__desc {
    color: $regular-text;
    margin-top: -6px;
    margin-bottom: 24px;
    max-width: 300px;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
  }

  &__exercise-count {
    color: $regular-text;
  }

  &__completed {
    color: $success;
  }

  &__icon {
    font-size: 32px;
  }
}
</style>
