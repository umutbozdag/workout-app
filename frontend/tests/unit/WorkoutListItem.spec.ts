import { mount } from "@vue/test-utils";
import WorkoutListItem from "@/components/WorkoutListItem.vue";
import { mockExerciseData } from "./generateMockData";

describe("WorkoutListItem.vue", () => {
  const wrapper = mount(WorkoutListItem, {
    propsData: {
      workoutId: mockExerciseData.workoutId,
      title: mockExerciseData.title,
      description: mockExerciseData.description,
      exerciseCount: 4,
      completed: mockExerciseData.completed,
    },
  });

  it("Props passes correctly", () => {
    expect(wrapper.props("workoutId")).toBe(mockExerciseData.workoutId);
    expect(wrapper.props("title")).toBe(mockExerciseData.title);
    expect(wrapper.props("description")).toBe(mockExerciseData.description);
    expect(wrapper.props("exerciseCount")).toBe(4);
    expect(wrapper.props("completed")).toBe(mockExerciseData.completed);
  });

  it("Contains 'workout-list-item' class", () => {
    expect(wrapper.classes()).toContain("workout-list-item");
  });

  it("Renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
