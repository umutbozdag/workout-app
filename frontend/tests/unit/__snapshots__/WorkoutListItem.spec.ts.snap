// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutListItem.vue Renders correctly 1`] = `
<div
  class="workout-list-item"
>
  <div
    class="workout-list-item__info"
  >
    <h3
      class="workout-list-item__title"
    >
      Exercise Title
    </h3>
     
    <p
      class="workout-list-item__desc"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </p>
     
    <p
      class="workout-list-item__completed"
    >
      
      Workout is completed
    
    </p>
     
    <div
      class="workout-list-item__exercise-count"
    >
      <!---->
       
        4 Exercises
       
      <!---->
    </div>
  </div>
   
  <i
    class="el-icon-right workout-list-item__icon"
  />
</div>
`;
